paddle/operators/target_assign_op.cu

```diff
@@ -0,0 +1,61 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/operators/target_assign_op.h"
+
+namespace paddle {
+namespace operators {
+
+template <typename T>
+__global__ void UpdateTargetLabelKernel(const int* neg_indices,
+                                        const size_t* lod, const int num,
+                                        const int num_prior_box,
+                                        const int background_label,
+                                        int* out_label, T* out_label_wt) {
+  int bidx = blockIdx.x;
+  int st = lod[bidx];
+  int ed = lod[bidx + 1];
+
+  for (int i = st + threadIdx.x; i < ed; i += blockDim.x) {
+    int id = neg_indices[i];
+    out_label[bidx * num_prior_box + id] = background_label;
+    out_label_wt[bidx * num_prior_box + id] = 1.;
```

Comment:
`bidx * num_prior_box` has appeared many times and it is inside a loop, so it should be optimized.

Reply:
Done.